@here/harp-mapview/lib/ThemeLoader.ts

```diff
@@ -460,69 +422,50 @@ export class ThemeLoader {
+        const themeStyles = getStyles(theme.styles);
         if (baseTheme.styles && theme.styles) {
-            const currentStyleSets = Object.keys(baseTheme.styles);
-            const incomingStyleSets = Object.keys(theme.styles);
-
-            styles = {};
-
-            currentStyleSets.forEach(styleSetName => {
-                const index = incomingStyleSets.indexOf(styleSetName);
-
-                if (index !== -1) {
-                    // merge the current and incoming styleset
-                    // and add the result to `styles`.
-
-                    const baseStyleSet = baseTheme.styles![styleSetName];
-
-                    const newStyleSet: StyleSet = [];
-                    const styleIdMap = new Map<string, number>();
-                    baseStyleSet.forEach(style => {
-                        if (typeof style.id === "string") {
-                            styleIdMap.set(style.id, newStyleSet.length);
-                        }
-                        newStyleSet.push(style);
-                    });
-
-                    const incomingStyleSet = theme.styles![styleSetName];
-                    incomingStyleSet.forEach(style => {
-                        if (typeof style.extends === "string" && styleIdMap.has(style.extends)) {
-                            // extends the existing style referenced by `style.extends`.
-                            const baseStyleIndex = styleIdMap.get(style.extends)!;
-                            const baseStyle = newStyleSet[baseStyleIndex];
-                            newStyleSet[baseStyleIndex] = { ...baseStyle, ...style } as any;
-                            newStyleSet[baseStyleIndex].extends = undefined;
-                            return;
-                        }
-
-                        if (typeof style.id === "string" && styleIdMap.has(style.id)) {
-                            // overrides the existing style with `id` equals to `style.id`.
-                            const styleIndex = styleIdMap.get(style.id)!;
-                            newStyleSet[styleIndex] = style;
-                            return;
-                        }
-
-                        newStyleSet.push(style);
-                    });
+            const newStyles: Styles = [];
+            const styleIdMap = new Map<string, number>();
+            baseStyles.forEach(style => {
+                if (typeof style.id === "string") {
+                    //multiple identical style.ids are not supported and will fall back to the
+                    //first occurence
+                    if (!styleIdMap.has(style.id)) {
+                        styleIdMap.set(style.id, newStyles.length);
+                    }
+                }
+                newStyles.push(style);
+            });
 
-                    styles[styleSetName] = newStyleSet;
+            themeStyles.forEach(style => {
+                if (typeof style.extends === "string" && styleIdMap.has(style.extends)) {
```

Comment:
Good point, I am not sure, if we should cover this in the scope of this task. But I see that the logic behind the merging of styles has not changed. Before styles with the same id would have been overrriden, and in case a style iwith and extends parameter matching an id in the baseTheme, it would have been extended. 
I agree, that both cases should be adjusted, the id override as mentioned above.
And the extends should either be done in ThemeLoader for all styles or moved completly to StyleSetEvaluator, but not partly here and partly there. 
Should we create a follow up ticket for this?

Reply:
Decided to revisit how style.extends works in a separate ticket.